### PR TITLE
auto superboom

### PR DESCRIPTION
**Pull request by x45k**
automatically places superboom when you left click on cracked stone bricks, but it has a 1 second swap delay, and a 1 second place delay so watchdog doesnt go funny mode (this can obv be lowered in the future, just there right now to be extra safe). i completely understand if this isnt wanted, because it is kinda a bannable area, but it shouldnt be now.

### CODE REVIEW

**Comment by x45k** [comment]
most recent commit allows you to change the amount of ticks yourself so disregard half of what i said in the description

File: odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import net.minecraftforge.event.entity.player.PlayerInteractEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object AutoBoom : Module(
+    name = "Auto Superboom",
+    description = "Places a superboom when you left-click.",
+    category = Category.DUNGEON,
+    tag = TagType.RISKY
+) {
+    private val minecraft: Minecraft = Minecraft.getMinecraft()
```
Comment (odtheking) [inline odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt:24]: Do not declare this again there is already a global variable for this
Don't copy paste random stuff into this mod

File: odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        // Delay for placing the item
+        runIn(placeDelay.toInt()) {
+            triggerRightClick()
+        }
+    }
+
+    private fun isLookingAtCrackedStoneBricks(pos: BlockPos): Boolean {
+        val block: Block = minecraft.theWorld.getBlockState(pos).block
```
Comment (odtheking) [inline odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt:51]: Use Odin's method for getting a block

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,22 +1,22 @@
-pluginManagement {
```
Comment (odtheking) [inline settings.gradle.kts:1]: Don't touch this file

File: odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    private val minecraft: Minecraft = Minecraft.getMinecraft()
+
+    private val placeDelay: Long by NumberSetting("Place Delay", 10L, 1, 20, unit = "ticks", description = "Placing Superboom delay in ticks.")
+    private val anyBlock: Boolean by BooleanSetting("Any Block", default = false, description = "Place Superboom on any block.")
+
+    @SubscribeEvent
+    fun onLeftClick(event: PlayerInteractEvent) {
+        if (event.action != PlayerInteractEvent.Action.LEFT_CLICK_BLOCK || (!anyBlock && !isLookingAtCrackedStoneBricks(event.pos))) return
+
+        val superboomSlot = getItemSlot("Superboom TNT") ?: return minecraft.thePlayer.addChatMessage(ChatComponentText("No item named 'Superboom TNT' could be found."))
```
Comment (odtheking) [inline odinclient/src/main/kotlin/me/odinclient/features/impl/dungeon/AutoBoom.kt:33]: Use Odin's own methods for sending a mod message 
Do not copy paste random stuff into this mod

File: settings.gradle.kts (hunk 3 above)
Comment (x45k) [reply settings.gradle.kts:1]: dont believe i did, never did touch it, pretty sure github is just going mentally insane. unless im blind, the 2 things are the exact same (mightve accidently added a space somewhere somehow? i have no clue)

**Pull request closed without merging**